erna/scripts/upload.py

```diff
@@ -74,17 +74,27 @@ def xml(config, fact_tools_version, name, comment, xml_file):
 
 @click.command()
 @click.option('--config', help='Yaml file containing database credentials')
+@click.option('--version', help='Set the version, if not the jar is executed to obtain it')
 @click.argument('jar-file')
-def jar(config, jar_file):
+def jar(config, version, jar_file):
 
     config = load_config(config)
 
     log.debug('Connecting to database')
     database.init(**config['processing_database'])
 
-    output = sp.check_output(['java', '-jar', jar_file]).decode()
-    version = re.match('.*Version ([0-9]+\.[0-9]+\.[0-9]+)', output).groups()[0]
-    log.info('Found FACT Tools version "{}"'.format(version))
+    output = sp.check_output(['java', '-jar', jar_file, '--version']).decode()
```

Comment:
You don't need to execute this if the version is supplied so move into the if, and while we are at it you could check whether the output is empty, as versions before our version stuff don't seem to output anything

Reply:
But this basically tests if the provided file is a valid jar, so I want to keep it above the if